src/main/java/io/confluent/idesidecar/restapi/credentials/TLSConfig.java

```diff
@@ -0,0 +1,304 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Null;
+import jakarta.validation.constraints.Size;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "SSL configuration")
+public record TLSConfig(
+    @Schema(description = "Whether to verify the server's SSL certificate. If set to false, "
+        + "the server's SSL certificate will not be verified. Defaults to true.")
+    @JsonProperty(value = "verify_ssl_certificates")
+    @Null
+    Boolean verifySslCertificates,
+    @Schema(description =
+        "The trust store configuration for authenticating the server's certificate.")
+    @JsonProperty(value = "truststore")
+    @NotNull
+    TrustStore truststore,
+    @Schema(description = "The key store configuration for "
+        + "authenticating the client to the server.",
+        nullable = true)
+    @JsonProperty(value = "keystore")
+    @Null
+    KeyStore keystore
+) {
+
+  public record TrustStore(
+      @Schema(description = "The path to the local trust store file. Required for authenticating "
+          + "the server's certificate.")
+      @JsonProperty(value = "path")
+      @Size(max = TRUSTSTORE_PATH_MAX_LEN)
+      @NotNull
+      String path,
+
+      @Schema(
+          description = "The password for the local trust store file. If a password is not set, "
+              + "trust store file configured will still be used, but integrity checking "
+              + "is disabled. A trust store password is not supported for PEM format.",
+          nullable = true
+      )
+      @JsonProperty(value = "password")
+      @Size(max = TRUSTSTORE_PASSWORD_MAX_LEN)
+      @Null
+      Password password,
+
+      @Schema(description = "The file format of the local trust store file", nullable = true)
+      @JsonProperty(value = "type")
+      @Null
+      StoreType type
```

Comment:
Maybe we should either have a default or require this to be not-null. As discussed earlier, the VS Code extension will likely set this anyway.

Reply:
> Maybe we should either have a default or require this to be not-null.

How about we do both? The default would be `JKS` but we still require it to be set.